// octacore-test/src/index.js
/*
 * Octacore
 * Welcome to the API documentation for our blockchain data service! Our service allows users to access and query data from various blockchain networks through a set of easy-to-use API endpoints. Whether you're looking for transaction history, block information, or on-chain analytics, our API has you covered. With our API, you can retrieve data from a variety of popular blockchain networks, including ethereum and polygon mainnets . Simply make a request to the appropriate endpoint with the necessary parameters, and you'll receive a response with the requested data in a convenient JSON format. We strive to make our API as user-friendly and reliable as possible, and are constantly working to expand our coverage and capabilities. If you have any questions or need assistance getting started, please don't hesitate to reach out to us. We hope that our API will be a valuable resource for your blockchain projects.
 *
 * OpenAPI spec version: 1.0
 * Contact: contact@octacore.io
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.31
 *
 * Do not edit the class manually.
 *
 */

(function(factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/ModelsAddressBalanceReq', 'model/ModelsAddressReq', 'model/ModelsBlockNumber', 'model/ModelsBlockNumberIndex', 'model/ModelsChains', 'model/ModelsContractAdressRequestBody', 'model/ModelsDomainReq', 'model/ModelsGetNFTRequestBody', 'model/ModelsGetSpenderAllowanceRequestBody', 'model/ModelsGetStorageAtReq', 'model/ModelsHash', 'model/ModelsHashIndex', 'model/ModelsIsOwnerInCollectionRequestBody', 'model/ModelsNonceReq', 'model/ModelsTxReq', 'api/AccountsApi', 'api/BlocksApi', 'api/ContractsApi', 'api/ENSApi', 'api/MetadataApi', 'api/NFTApi', 'api/TransactionsApi'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('./ApiClient'), require('./model/ModelsAddressBalanceReq'), require('./model/ModelsAddressReq'), require('./model/ModelsBlockNumber'), require('./model/ModelsBlockNumberIndex'), require('./model/ModelsChains'), require('./model/ModelsContractAdressRequestBody'), require('./model/ModelsDomainReq'), require('./model/ModelsGetNFTRequestBody'), require('./model/ModelsGetSpenderAllowanceRequestBody'), require('./model/ModelsGetStorageAtReq'), require('./model/ModelsHash'), require('./model/ModelsHashIndex'), require('./model/ModelsIsOwnerInCollectionRequestBody'), require('./model/ModelsNonceReq'), require('./model/ModelsTxReq'), require('./api/AccountsApi'), require('./api/BlocksApi'), require('./api/ContractsApi'), require('./api/ENSApi'), require('./api/MetadataApi'), require('./api/NFTApi'), require('./api/TransactionsApi'));
  }
}(function(ApiClient, ModelsAddressBalanceReq, ModelsAddressReq, ModelsBlockNumber, ModelsBlockNumberIndex, ModelsChains, ModelsContractAdressRequestBody, ModelsDomainReq, ModelsGetNFTRequestBody, ModelsGetSpenderAllowanceRequestBody, ModelsGetStorageAtReq, ModelsHash, ModelsHashIndex, ModelsIsOwnerInCollectionRequestBody, ModelsNonceReq, ModelsTxReq, AccountsApi, BlocksApi, ContractsApi, ENSApi, MetadataApi, NFTApi, TransactionsApi) {
  'use strict';
  /**
   * Welcome_to_the_API_documentation_for_our_blockchain_data_serviceOur_service_allows_users_to_access_and_query_data_from_various_blockchain_networks_through_a_set_of_easy_to_use_API_endpoints__Whether_youre_looking_for_transaction_history_block_information_or_on_chain_analytics_our_API_has_you_covered_With_our_API_you_can_retrieve_data_from_a_variety_of_popular_blockchain_networks_including_ethereum_and_polygon_mainnets___Simply_make_a_request_to_the_appropriate_endpoint_with_the_necessary_parameters_and_youll_receive_a_response_with_the_requested_data_in_a_convenient_JSON_format_We_strive_to_make_our_API_as_user_friendly_and_reliable_as_possible_and_are_constantly_working_to_expand_our_coverage_and_capabilities__If_you_have_any_questions_or_need_assistance_getting_started_please_dont_hesitate_to_reach_out_to_us__We_hope_that_our_API_will_be_a_valuable_resource_for_your_blockchain_projects_.<br>
   * The <code>index</code> module provides access to constructors for all the classes which comprise the public API.
   * <p>
   * An AMD (recommended!) or CommonJS application will generally do something equivalent to the following:
   * <pre>
   * var Octacore = require('index'); // See note below*.
   * var xxxSvc = new Octacore.XxxApi(); // Allocate the API class we're going to use.
   * var yyyModel = new Octacore.Yyy(); // Construct a model instance.
   * yyyModel.someProperty = 'someValue';
   * ...
   * var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
   * ...
   * </pre>
   * <em>*NOTE: For a top-level AMD script, use require(['index'], function(){...})
   * and put the application logic within the callback function.</em>
   * </p>
   * <p>
   * A non-AMD browser application (discouraged) might do something like this:
   * <pre>
   * var xxxSvc = new Octacore.XxxApi(); // Allocate the API class we're going to use.
   * var yyy = new Octacore.Yyy(); // Construct a model instance.
   * yyyModel.someProperty = 'someValue';
   * ...
   * var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
   * ...
   * </pre>
   * </p>
   * @module index
   * @version 1.0
   */
  var exports = {
    /**
     * The ApiClient constructor.
     * @property {module:ApiClient}
     */
     Core: ApiClient,
    /**
     * The ModelsAddressBalanceReq model constructor.
     * @property {module:model/ModelsAddressBalanceReq}
     */
    ModelsAddressBalanceReq: ModelsAddressBalanceReq,
    /**
     * The ModelsAddressReq model constructor.
     * @property {module:model/ModelsAddressReq}
     */
    ModelsAddressReq: ModelsAddressReq,
    /**
     * The ModelsBlockNumber model constructor.
     * @property {module:model/ModelsBlockNumber}
     */
    ModelsBlockNumber: ModelsBlockNumber,
    /**
     * The ModelsBlockNumberIndex model constructor.
     * @property {module:model/ModelsBlockNumberIndex}
     */
    ModelsBlockNumberIndex: ModelsBlockNumberIndex,
    /**
     * The ModelsChains model constructor.
     * @property {module:model/ModelsChains}
     */
    ModelsChains: ModelsChains,
    /**
     * The ModelsContractAdressRequestBody model constructor.
     * @property {module:model/ModelsContractAdressRequestBody}
     */
    ModelsContractAdressRequestBody: ModelsContractAdressRequestBody,
    /**
     * The ModelsDomainReq model constructor.
     * @property {module:model/ModelsDomainReq}
     */
    ModelsDomainReq: ModelsDomainReq,
    /**
     * The ModelsGetNFTRequestBody model constructor.
     * @property {module:model/ModelsGetNFTRequestBody}
     */
    ModelsGetNFTRequestBody: ModelsGetNFTRequestBody,
    /**
     * The ModelsGetSpenderAllowanceRequestBody model constructor.
     * @property {module:model/ModelsGetSpenderAllowanceRequestBody}
     */
    ModelsGetSpenderAllowanceRequestBody: ModelsGetSpenderAllowanceRequestBody,
    /**
     * The ModelsGetStorageAtReq model constructor.
     * @property {module:model/ModelsGetStorageAtReq}
     */
    ModelsGetStorageAtReq: ModelsGetStorageAtReq,
    /**
     * The ModelsHash model constructor.
     * @property {module:model/ModelsHash}
     */
    ModelsHash: ModelsHash,
    /**
     * The ModelsHashIndex model constructor.
     * @property {module:model/ModelsHashIndex}
     */
    ModelsHashIndex: ModelsHashIndex,
    /**
     * The ModelsIsOwnerInCollectionRequestBody model constructor.
     * @property {module:model/ModelsIsOwnerInCollectionRequestBody}
     */
    ModelsIsOwnerInCollectionRequestBody: ModelsIsOwnerInCollectionRequestBody,
    /**
     * The ModelsNonceReq model constructor.
     * @property {module:model/ModelsNonceReq}
     */
    ModelsNonceReq: ModelsNonceReq,
    /**
     * The ModelsTxReq model constructor.
     * @property {module:model/ModelsTxReq}
     */
    ModelsTxReq: ModelsTxReq,
    /**
     * The AccountsApi service constructor.
     * @property {module:api/AccountsApi}
     */
     Accounts: AccountsApi,
     /**
      * The BlocksApi service constructor.
      * @property {module:api/BlocksApi}
      */
     Blocks: BlocksApi,
     /**
      * The ContractsApi service constructor.
      * @property {module:api/ContractsApi}
      */
     Contracts: ContractsApi,
     /**
      * The ENSApi service constructor.
      * @property {module:api/ENSApi}
      */
     ENS: ENSApi,
     /**
      * The MetadataApi service constructor.
      * @property {module:api/MetadataApi}
      */
     Metadata: MetadataApi,
     /**
      * The NFTApi service constructor.
      * @property {module:api/NFTApi}
      */
     NFT: NFTApi,
     /**
      * The TransactionsApi service constructor.
      * @property {module:api/TransactionsApi}
      */
     Transactions: TransactionsApi,
  };

  return exports;
}));
